java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1067,31 +872,24 @@ public StatementExpressionListTreeImpl newForInitDeclaration(ModifiersTreeImpl m
     }
 
     StatementExpressionListTreeImpl result = new StatementExpressionListTreeImpl(variables);
-    result.prependChildren(modifiers, (AstNode) type, variables);
 
     return result;
   }
 
-  public StatementExpressionListTreeImpl newStatementExpressions(ExpressionTree expression, Optional<List<AstNode>> rests) {
-    List<AstNode> children = Lists.newArrayList();
+  public StatementExpressionListTreeImpl newStatementExpressions(ExpressionTree expression, Optional<List<Tuple<InternalSyntaxToken, ExpressionTree>>> rests) {
     ImmutableList.Builder<StatementTree> statements = ImmutableList.builder();
 
     ExpressionStatementTreeImpl statement = new ExpressionStatementTreeImpl(expression, null);
     statements.add(statement);
-    children.add(statement);
 
     if (rests.isPresent()) {
-      for (AstNode rest : rests.get()) {
-        children.add(rest.getFirstChild());
-
-        statement = new ExpressionStatementTreeImpl((ExpressionTree) rest.getLastChild(), null);
+      for (Tuple<InternalSyntaxToken, ExpressionTree> rest : rests.get()) {
+        statement = new ExpressionStatementTreeImpl(rest.second(), null);
         statements.add(statement);
-        children.add(statement);
       }
     }
 
     StatementExpressionListTreeImpl result = new StatementExpressionListTreeImpl(statements.build());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)